src/dialogs/more-info/ha-more-info-tab-history.ts

```diff
@@ -0,0 +1,170 @@
+import {
+  css,
+  customElement,
+  html,
+  internalProperty,
+  LitElement,
+  property,
+  PropertyValues,
+  TemplateResult,
+} from "lit-element";
+import { classMap } from "lit-html/directives/class-map";
+import { computeStateDomain } from "../../common/entity/compute_state_domain";
+import "../../components/ha-circular-progress";
+import "../../components/state-history-charts";
+import { getRecentWithCache } from "../../data/cached-history";
+import { HistoryResult } from "../../data/history";
+import { getLogbookData, LogbookEntry } from "../../data/logbook";
+import "../../panels/logbook/ha-logbook";
+import { haStyleDialog } from "../../resources/styles";
+import { HomeAssistant } from "../../types";
+
+@customElement("ha-more-info-tab-history")
+export class MoreInfoTabHistoryDialog extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property() public entityId!: string;
+
+  @internalProperty() private _stateHistory?: HistoryResult;
+
+  @internalProperty() private _isLoading = false;
+
+  private _entries: LogbookEntry[] = [];
+
+  private _persons = {};
+
+  private _historyRefreshInterval?: number;
+
+  protected render(): TemplateResult {
+    if (!this.entityId) {
+      return html``;
+    }
+    const entityId = this.entityId;
+    const stateObj = this.hass.states[entityId];
+
+    if (!stateObj) {
+      return html``;
+    }
+
+    return html`
+      ${this._isLoading
+        ? html`
+            <ha-circular-progress
+              active
+              alt=${this.hass.localize("ui.common.loading")}
+            ></ha-circular-progress>
+          `
+        : html`
+            <state-history-charts
+              up-to-now
+              .hass=${this.hass}
+              .historyData=${this._stateHistory}
+            ></state-history-charts>
+            <ha-logbook
+              narrow
+              no-click
+              no-icon
+              no-name
+              class=${classMap({ "no-entries": !this._entries.length })}
+              .hass=${this.hass}
+              .entries=${this._entries}
+              .userIdToName=${this._persons}
+            ></ha-logbook>
+          `}
+    `;
+  }
+
+  protected updated(changedProps: PropertyValues): void {
+    super.updated(changedProps);
+    if (!this.entityId) {
+      clearInterval(this._historyRefreshInterval);
+    }
+
+    if (changedProps.has("entityId")) {
+      this._refreshData();
+    }
+  }
+
+  private async _refreshData(): Promise<void> {
+    this._isLoading = true;
+    this._fetchPersonNames();
```

Comment:
We only need to do this when person entities changed

Reply:
How do we know they changed?

Reply:
Check for state changes, but might not be worth the trouble, never mind.

Reply:
Should I just fetch it once on first updated and not worry about it?